web_analytics.go

```diff
@@ -0,0 +1,393 @@
+type WebAnalyticsIDResponse struct {
+	Response
+	Result struct {
+		ID string `json:"id"`
+	} `json:"result"`
+}
+
+// WebAnalyticsSiteTagResponse is the API response, containing a single ID.
+type WebAnalyticsSiteTagResponse struct {
+	Response
+	Result struct {
+		SiteTag string `json:"site_tag"`
+	} `json:"result"`
+}
+
+type CreateWebAnalyticsSiteParams struct {
+	// Host is the host to measure traffic for.
+	Host string `json:"host,omitempty"`
+	// ZoneTag is the zone tag to measure traffic for.
+	ZoneTag string `json:"zone_tag,omitempty"`
+	// AutoInstall defines whether Cloudflare will inject the JS snippet automatically for orange-clouded sites.
+	AutoInstall bool `json:"auto_install"`
+}
+
+// CreateWebAnalyticsSite creates a new Web Analytics Site for an Account.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-create-site
+func (api *API) CreateWebAnalyticsSite(ctx context.Context, rc *ResourceContainer, params CreateWebAnalyticsSiteParams) (*WebAnalyticsSite, error) {
+	if params.Host == "" && params.ZoneTag == "" {
+		return nil, ErrMissingWebAnalyticsSiteHost
+	}
+	uri := fmt.Sprintf("/accounts/%s/rum/site_info", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
+	if err != nil {
+		return nil, err
+	}
+	var r WebAnalyticsSiteResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result, nil
+}
+
+type ListWebAnalyticsSitesParams struct {
+	// Page offset for pagination.
+	Page int
+	// Items per page limit for pagination.
+	PerPage int
+	// Property to order Sites by, "host" or "created".
+	OrderBy string
+}
+
+// ListWebAnalyticsSites returns all Web Analytics Sites for an Account.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-list-sites
+func (api *API) ListWebAnalyticsSites(ctx context.Context, rc *ResourceContainer, params ListWebAnalyticsSitesParams) ([]WebAnalyticsSite, *ResultInfo, error) {
+	v := url.Values{}
+	if params.Page > 0 {
+		v.Set("page", strconv.Itoa(params.Page))
+	}
+	if params.PerPage > 0 {
+		v.Set("per_page", strconv.Itoa(params.PerPage))
+	}
+	if params.OrderBy != "" {
+		v.Set("order_by", params.OrderBy)
+	}
+	uri := fmt.Sprintf("/accounts/%s/rum/site_info/list?"+v.Encode(), rc.Identifier)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	var r WebAnalyticsSitesResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return r.Result, &r.ResultInfo, nil
+}
+
+type WebAnalyticsSiteParams struct {
+	SiteTag string
+}
+
+// WebAnalyticsSite fetches detail about one Web Analytics Site for an Account.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-get-site
+func (api *API) WebAnalyticsSite(ctx context.Context, rc *ResourceContainer, params WebAnalyticsSiteParams) (*WebAnalyticsSite, error) {
```

Comment:
for fetching single resources, we should favour `Get` as a prefix.

```suggestion
func (api *API) GetWebAnalyticsSite(ctx context.Context, rc *ResourceContainer, params GetWebAnalyticsSiteParams) (*WebAnalyticsSite, error) {
```